client/my-sites/domains/domain-management/dns/index.jsx

```diff
@@ -109,6 +100,18 @@ class Dns extends React.Component {
 		);
 	}
 
+	render() {
+		const { showPlaceholder, selectedDomainName, selectedSite } = this.props;
+
+		return (
+			<Fragment>
+				<QuerySiteDomains siteId={ selectedSite.ID } />
```

Comment:
It's probably a good idea to rewrite this query component the same way we did for the `QueryDomainDns`. Now it fetches domains every time we open the DNS Editor, and it's kind of an overkill. Tho, we'd prolly need a way to force refresh it then, as some paths shouldn't do caching.

An idea for later @Automattic/cobalt.

Reply:
The current behavior is that we query site domains every time any domains screen is opened, and also when switching between different screens (e.g., DNS, nameservers, transfer, ...). The request is _not_ issued only when another one is already in flight (`isRequestingSiteDomains`).

What you probably mean is to never refetch once a domains list has been loaded from server? I.e., introduce the `hasLoadedFromServer` flag?

Reply:
> What you probably mean is to never refetch once a domains list has been loaded from server? I.e., introduce the hasLoadedFromServer flag?

That is right. :) Some screens, like the `/domains` should refetch always. DNS one tho, I don't think it's used for anything except `selectedDomain`.